### PR TITLE
Restore private repo in some features

### PR DESCRIPTION
**Pull request by yakov116**
Follows https://github.com/sindresorhus/refined-github/commit/9ddcd95d766d26af8fe6362c65f32370464c5bd2#r39608686

### CODE REVIEW

**Comment by fregante** [comment]
Can you add a comment with the two examples? Otherwise this is bound to get “fixed” again

**Comment by fregante** [comment]
Better yet, make the regex more specific since it currently breaks if a branch or repo name contains `.atom`

**Comment by yakov116** [comment]
@fregante done.
I force-pushed since I reopened a very old branch (from my last pr)

**Comment by fregante** [comment]
> Better yet, make the regex more specific since it currently breaks if a branch or repo name contains `.atom`

I prefer this

File: source/github-helpers/index.ts [hunk 1]
```diff
@@ -18,15 +18,16 @@ export const getDiscussionNumber = (): string | undefined => {
...
+Example tag content on private repositories https://github.com/private/private/commits/master.atom?token=AEAXKWNRHXA2XJ2ZWCMGUUN44LM62
 */
 export const getCurrentBranch = (): string => {
 	return select<HTMLLinkElement>('[type="application/atom+xml"]')!
 		.href
 		.split('/')
 		.slice(6) // Drops the initial https://host/user/repo/route/ part
 		.join('/')
-		.replace(/\.atom$/, '');
+		.replace(/\.atom[^.]*$/, '');
```
Comment (fregante) [inline source/github-helpers/index.ts:30]: I meant something like this
```suggestion
		.replace(/\.atom(\?token=.+)?$/, '');
```

**Comment by sindresorhus** [comment]
Feels like it would be more future-proof to use `URL` to clear the query string:

```js
const x = new URL('https://github.com/private/private/commits/master.atom?token=AEAXKWNRHX');
x.search = '';
console.log(x.pathname);
```

**Comment by yakov116** [comment]
Or even
```
const x = new URL('https://github.com/private/private/commits/master.atom?token=AEAXKWNRHX').pathname.split('/')....;
```

Then there is no need to even clear the search

**Pull request merged**